### PR TITLE
SSML Models Implementation

### PR DESCRIPTION
**Pull request by stompsjo**
This PR introduces much of the code used for my research comparing several different semi-supervised machine learning models (SSML). I wrote and used these models quite separate from how I have used RadClass in the past (i.e. I ran experimental data through RadClass and H0 to get observations I could use for training and testing ML models separately in a downstream analysis).

This raises a design question regarding how these models and scripts should be integrated into RadClass. Should they be made into a post_analysis object? Should they remain separate, thereby making the user have a script that both runs RadClass for data processing and SSML models for training and testing?

- Another note that popped into my head: should I include a `.load` method for each model class? This may remove one manual step a user would have to take to use `joblib` for loading a written model and storing as a model class.

### CODE REVIEW

**Comment by stompsjo** [comment]
One problem to note: creating tests for neural networks has been tricky. Ideally, these tests should not take several minutes to train and test a method/model. But if we want meaningful test-cases, we will want to test meaningful neural network architectures (which take longer to train).

I will be thinking of a compromise...

**Comment by stompsjo** [comment]
The most recent commit changes each `fresh_start` method for each ML model to use the parent class `train` method rather than copying its logic outright. This should help to make each class concise and more comprehensible as a whole.

**Comment by gonuke** [comment]
Split into more PRs?

File: scripts/utils.py [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+    print(pca.singular_values_)
+    print(pca.components_)
+
+    alph = ["A", "B", "C", "D", "E", "F", "G", "H",
+            "I", "J", "K", "L", "M", "N", "O", "P",
+            "Q", "R", "S", "T", "U", "V", "W", "X",
+            "Y", "Z"]
+    jobs = alph[:n]
+
+    fig, axes = plt.subplots(n, n, figsize=(15, 15))
```
Comment (stompsjo) [inline scripts/utils.py:280]:
```suggestion
    fig, axes = plt.subplots(n, n, figsize=(3*n, 3*n))
```

**Comment by stompsjo** [comment]
This PR has been split into #41, #42, #44, and #45

**Pull request closed without merging**